test/e2e/smoke_test.go

```diff
@@ -0,0 +1,141 @@
+//+build e2e
+
+package e2e
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"regexp"
+	"strconv"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	templatev1 "github.com/openshift/api/template/v1"
+	routev1client "github.com/openshift/client-go/route/clientset/versioned/typed/route/v1"
+	templatev1client "github.com/openshift/client-go/template/clientset/versioned/typed/template/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/watch"
+	corev1client "k8s.io/client-go/kubernetes/typed/core/v1"
+)
+
+const GOOD_TEMPLATE = "nodejs-mongodb-example"
+const BAD_TEMPLATE = "non-existent"
+const REGEX = `\<span.*id=\"count-value\"\>(?P<counter>\d+)\<\/span\>`
+
+var _ = Describe("Openshift on Azure end user e2e tests [SmokeTest]", func() {
+	defer GinkgoRecover()
+
+	Context("When the template exists", func() {
```

Comment:
I would expect to see this test as part of the end_user_test.go file since it's an e2e test that should work for end users.